Modules/Core/ImageAdaptors/test/itkVectorImageTest.cxx

```diff
@@ -85,6 +85,7 @@ testVectorImageAdaptor(typename TAdaptor::Pointer &
   {
     using CovariantVectorType = itk::CovariantVector<double, VDimension>;
     CovariantVectorType input;
+    input.Fill(0);
```

Comment:
`input.Fill(0)` is fine, although it could also be achieved by C++11 empty initialization-list: `CovariantVectorType input{}`.

Reply:
I can't grok through all the constructors and C++ POD/aggregate initialization rules to know how to construct all the different array-like ITK class.

Hrmm... The class is not POD, and it has private data so it's not aggregate initialization. `{}` can't be calling the default constructor, because that should have been called with the prior syntax...I really don't know what `{}` is calling, or why.

Reply:
In general, I think a default-constructible class `T` should be designed in such a way that a variable `var1` is properly initialized by:

    T var1{};  // Should be properly initialized.

That also works well when `T = int`, it's a fundamental part of C++11 _uniform initialization_.

Whether or not a variable `var2` is properly initialized without the empty initializer-list depends on the type `T`, of course.

    T var2;  // May not be initialized.

My 2 cents

Reply:
I used `{0.0}` for the initialization to make it explicit its value is zero.

@N-Dekker It was mistaken, the `CovariantVector` is POD as checked by this [nice test](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/test/itkCommonTypeTraitsGTest.cxx#L52). I know understand what is going on with the constructors and the syntax.